src/common/ad_mem.c

```diff
@@ -1668,7 +2019,7 @@ ad_reserve_addr(struct ad_blob *blob, int type, daos_size_t size,
 		if (rc) {
 			D_DEBUG(DB_TRACE, "Failed to load arena %u: %d\n", id, rc);
 			/* fall through and create a new one */
-		} else if (arena->ar_full) {
+		} if (!arena2heap_node(arena)->mh_active) {
```

Comment:
(style)  trailing statements should be on next line (or did you mean 'else if'?)